pkg/monitor/azure/nsg/nsg_test.go

```diff
@@ -547,3 +566,99 @@ func wait(wg *sync.WaitGroup, done chan<- any) {
 	wg.Wait()
 	done <- nil
 }
+
+func isOfType[T any](mon monitoring.Monitor) bool {
```

Comment:
Nice to see more generics :) 

2 things:
- This type of 1-2 line function seems a bit Java-y to me (not a bad thing). Is it common to do this type of thing in Go too?
- Potentially of interest/curious to hear your thoughts on this: https://github.com/Azure/ARO-RP/pull/3376#discussion_r1478472271

Reply:
I get you. I like this and agree it's the best choice. Thanks for giving context :)